### PR TITLE
doc: add main doc

### PR DESCRIPTION
**Pull request by moreirathomas**
## Description

Write main doc. To be updated when new routes for statistics handling and auth are implemented.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,143 @@
...
+# alias to:
+# golangci-lint run
+```
+
+### Test
+
+Run all tests:
+
+```sh
+make test
```
Comment (GregoryAlbouy) [inline README.md]:
```suggestion
make tests
```

File: README.md [hunk 2]
```diff
@@ -1 +1,143 @@
...
+# alias to:
+# go test -v -timeout 30s ./...
+```
+
+Run a specific test passing `t` to specify a test and `p` to specify a package (parameters are independent):
+
+```sh
+make test t=TestCompute p=pkg/golastic
+# alias to:
+# go test -v -timeout 30s -run TestCompute ./stats
```
Comment (GregoryAlbouy) [inline README.md]: No `pkg/golastic` nor `./stats` here

File: README.md [hunk 3]
```diff
@@ -1 +1,143 @@
...
+
+```txt
+POST /v1/reports
+```
+
+#### Parameters
+
+| Name     | Type                   | In   | Description                                            |
+| -------- | ---------------------- | ---- | ------------------------------------------------------ |
+| `report` | bytes (`gob` encoding) | body | The raw report data from the generated by the `runner` |
```
Comment (c1-ra) [inline README.md]: I do not understand the description really well :thinking:

**Review by c1-ra — CHANGES_REQUESTED** [review]
Looks good except for my comment!
I will add everything concerning the database after you have merged it.

**Review by c1-ra — APPROVED** [review]

**Review by GregoryAlbouy — APPROVED** [review]

**Review by c1-ra — DISMISSED** [review]
Nice, thanks :)

File: README.md [hunk 4]
```diff
@@ -1 +1,224 @@
...
+  "deciles": [
+    152005288, 262461226, 263282513, 265239215, 265682968, 267743397, 268441975,
+    268631834, 269033707
+  ],
+  "code1xx": 0,
+  "code2xx": 1000,
+  "code3xx": 0,
+  "code4xx": 0,
+  "code5xx": 0
+}
```
Comment (c1-ra) [inline README.md]: My bad, it is actually not exactly what is returned.
I have sent you some examples.

**Review by c1-ra — CHANGES_REQUESTED** [review]
Sorry, JSON response to `/stats/id` is actually slightly different.
Here is an example:
```
{
  "statsDescriptor": {
    "id": "azdiIUa8uq7zdn1",
    "tag": "my-custome-tag",
    "finishedAt": "2022-03-01T16:36:38Z"
  },
  "codestats": {
    "code1xx": 0,
    "code2xx": 1000,
    "code3xx": 0,
    "code4xx": 0,
    "code5xx": 0
  },
  "timestats": {
    "min": x,
    "max": x,
    "mean": x,
    "median": x,
    "standardDeviation": x,
    "deciles": [
      x,
      x,
      x,
      x,
      x,
      x,
      x,
      x,
      x
    ]
  }
}
```

File: README.md [hunk 5]
```diff
@@ -165,23 +165,29 @@ Status: 200 OK
...
-  "code1xx": 0,
-  "code2xx": 1000,
-  "code3xx": 0,
-  "code4xx": 0,
-  "code5xx": 0
+  "descriptor": {
+    "id": "lwkaFmtuCoeKSFhbndTC",
+    "tag": "my-custom-tag",
+    "finishedAt": "2022-02-27T19:54:19.019717Z",
+    "min": 152005288
```
Comment (c1-ra) [inline README.md]: "min" is a little too high :stuck_out_tongue:
Otherwise, it is good! :+1:

Comment (moreirathomas) [reply README.md]: woops

**Review by c1-ra — CHANGES_REQUESTED** [review]
Nice!
Just need to change `min` and I think we are good to go.

**Review by c1-ra — APPROVED** [review]
:+1:

**Pull request merged**